### PR TITLE
docs: v5 migration

### PR DESCRIPTION
**Pull request by MadCcc**
[[中文版模板 / Chinese template](https://github.com/ant-design/ant-design/blob/master/.github/PULL_REQUEST_TEMPLATE/pr_cn.md)]

### 🤔 This is a ...

- [ ] New feature
- [ ] Bug fix
- [x] Site / documentation update
- [ ] Demo update
- [ ] Component style update
- [ ] TypeScript definition update
- [ ] Bundle size optimization
- [ ] Performance optimization
- [ ] Enhancement feature
- [ ] Internationalization
- [ ] Refactoring
- [ ] Code style optimization
- [ ] Test Case
- [ ] Branch merge
- [ ] Other (about what?)

### 📝 Changelog

| Language   | Changelog |
| ---------- | --------- |
| 🇺🇸 English |       -    |
| 🇨🇳 Chinese |     -      |

### ☑️ Self-Check before Merge

⚠️ Please check all items below before review. ⚠️

- [x] Doc is updated/provided or not needed
- [x] Demo is updated/provided or not needed
- [x] TypeScript definition is updated/provided or not needed
- [x] Changelog is provided or not needed

### CODE REVIEW

File: docs/react/migration-v5.zh-CN.md [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+- 部分组件内间距调整。
+- 整体去线框化。
+
+### 技术调整
+
+- 弃用 less，采用 CSS-in-JS，更好地支持动态主题。底层使用 [@ant-design/cssinjs](https://github.com/ant-design/cssinjs) 作为解决方案。
+  - 所有 less 文件全部废弃，less 变量不再支持透出。
+  - 产物中不再包含 css 文件。由于 CSS-in-JS 支持按需引入，原本的 `antd/dist/antd.css` 也已经废弃，如果需要重置一些基本样式请引入 `antd/dist/reset.css`。
+- 废弃 css variables 以及在此之上构筑的动态主题方案。
+- 废弃 `lib` 产物，只提供 `dist` 和 `es` 产物。
```
Comment (afc163) [inline docs/react/migration-v5.zh-CN.md:28]: 改成『移除』，『废弃』代表能用但不推荐。

File: docs/react/migration-v5.zh-CN.md [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+- Upload List 结构变化。
+- Notification
+  - 静态方法不在允许在 `open` 中动态设置 `prefixCls` `maxCount` `top` `bottom` `getContainer`，Notification 静态方法现在将只有一个实例。如果需要不同配置，请使用 `useNotification`。
+  - `close` 改名为 `destroy` 和 message 保持一致。
+- Drawer
+  - `style` & `className` 迁移至 Drawer Panel 中，原属性替换为 `rootClassName` 和 `rootStyle`。
+
+#### 组件重构与废弃
+
+- PageHeader 和 Comment 组件在 ant-design 中废弃，移至 [pro-components](https://github.com/ant-design/pro-components) 中维护，如果仍需使用可以从兼容包中引入。
```
Comment (afc163) [inline docs/react/migration-v5.zh-CN.md:64]: 要写出如何从兼容包引入，使用 diff block 来描述。

File: docs/react/migration-v5.en-US.md [hunk 3]
```diff
@@ -0,0 +1,95 @@
...
+```diff
+- import { PageHeader, Comment, Input, Button } from 'antd';
++ import { PageHeader, Comment } from '@ant-design/compatible';
++ import '@ant-design/compatible/assets/index.css';
++ import { Input, Button } from 'antd';
+
+  const App: React.FC = () => (
+    <div>
+      <PageHeader />
+      <Comment />
```
Comment (afc163) [inline docs/react/migration-v5.en-US.md:86]: 不是 https://github.com/ant-design/pro-components 么？

**Comment by afc163** [comment]
使用 umi 和 babel-plugin-import 的情况也需要单独写一下如何改。

**Comment by afc163** [comment]
codemod 啥时候搞

File: docs/react/migration-v5.en-US.md (hunk 3 above)
Comment (MadCcc) [reply docs/react/migration-v5.en-US.md:136]: 记得之前说兼容包里也会提供的

**Comment by MadCcc** [comment]
> codemod 啥时候搞

codemod 应该等 api 都改掉吧

**Comment by afc163** [comment]
升级文档里可以先把 codemod 的预期用法也都写了。

**Comment by MadCcc** [comment]
1. Umi 插件部分待更新插件后完善
2. codemod 部分待组件 API 修改完成后完善

**Comment by afc163** [comment]
1. v4 主题包，提供给不想有样式变化的升级应用。
2. less 里用了 antd v4 变量的情况，如何兼容和升级。

**Comment by reichhartd** [comment]
For us it would also be very interesting how the global AntDesign variables can be modified and accessed in the code.

Also the use of `@ant-design/cssinjs` is not yet well documented.

**Comment by MadCcc** [comment]
先合了，给到社区继续一起补充

**Pull request merged**